File: detox/test/ios/ReactModules/ShakeEventEmitter.m
//
//  ShakeEventEmitter.m (example)
//  Created by Asaf Korem (Wix.com) on 2025.
//


#import "ShakeEventEmitter.h"

@implementation ShakeEventEmitter {
    BOOL hasListeners;
}

static ShakeEventEmitter *sharedInstance = nil;

RCT_EXPORT_MODULE();

+ (instancetype)sharedInstance {
    return sharedInstance;
}

- (instancetype)init {
    if (self = [super init]) {
        sharedInstance = self;
    }
    return self;
}

+ (BOOL)requiresMainQueueSetup {
    return YES;
}

- (NSArray<NSString *> *)supportedEvents {
    return @[@"ShakeEvent"];
}

- (void)startObserving {
    hasListeners = YES;
}

- (void)stopObserving {
    hasListeners = NO;
}

- (void)handleShake {
    if (hasListeners) {
        [self sendEventWithName:@"ShakeEvent" body:nil];
    }
}

@end

File: detox/test/ios/ReactModules/ShakeEventEmitter.swift
//
//  ShakeEventEmitter.swift (example)
//  Created by Asaf Korem (Wix.com) on 2024.
//

import Foundation
import React

@objc(ShakeEventEmitter)
class ShakeEventEmitter: RCTEventEmitter {

    static var reactInstance: ShakeEventEmitter? = nil

    override init() {
        super.init()
        ShakeEventEmitter.reactInstance = self
    }

    // MARK: - RCTEventEmitter Overrides

    override static func requiresMainQueueSetup() -> Bool {
        return true
    }

    override func supportedEvents() -> [String]! {
        return ["ShakeEvent"]
    }

    // MARK: - Public Methods

    @objc
    func handleShake() {
        sendEvent(withName: "ShakeEvent", body: nil)
    }
}
